messaging/src/main/java/org/axonframework/messaging/FluxMessageStream.java

```diff
@@ -34,10 +41,17 @@
 class FluxMessageStream<M extends Message<?>> implements MessageStream<M> {
 
     private final Flux<Entry<M>> source;
+    private final BlockingQueue<Entry<M>> readAhead = new LinkedBlockingQueue<>(5);
```

Comment:
Why does the queue have size of 5?

Reply:
Theoretically, it would need only 1, as it should only read a single item ahead. I figured theory and reality don't often align very well, so I picked a small number. Since it's a linked queue, it doesn't really matter much in terms of data allocation.